polars/polars-time/src/chunkedarray/utf8/mod.rs

```diff
@@ -517,10 +517,24 @@ pub trait Utf8Methods: AsUtf8 {
                             opt_s.and_then(|s| {
                                 if cache {
                                     *cache_map.entry(s).or_insert_with(|| {
-                                        NaiveDateTime::parse_from_str(s, &fmt).ok().map(func)
+                                        NaiveDateTime::parse_from_str(s, &fmt)
+                                            .ok()
+                                            .map(func)
+                                            .or_else(|| {
+                                                NaiveDate::parse_from_str(s, &fmt).ok().map(|nd| {
+                                                    func(nd.and_hms_opt(0, 0, 0).unwrap())
+                                                })
+                                            })
```

Comment:
I've made this the same as it is in

https://github.com/pola-rs/polars/blob/1fe6430600a1dd2b085c7201833059066a1092c1/polars/polars-time/src/chunkedarray/utf8/infer.rs#L184-L193

Reply:
Maybe we can use that function, or factor out the code into a function?

Reply:
yes, that's better, thanks!